src/extension/infer.rs

```diff
@@ -317,6 +317,12 @@ impl UnificationContext {
                         m_output,
                         node_type.op_signature().extension_reqs,
                     );
+                    if matches!(
```

Comment:
I've tried a few things here - see e.g. commit 59eb260 and [this comment](https://github.com/CQCL/hugr/pull/622/files#r1372772845). Think this is better than that...

but a better-still way (???) might be to define `OpType::default_extensions() -> Option<ExtensionSet>` and then make `add_op_xxxx` use not `open_extensions` but `default`. (Where `default_extensions` is `None` i.e. open for most OpTypes, but `ExtensionSet::new()` i.e. pure for Alias/Function/FuncDefn.) Thoughts?

Reply:
(Could be good for a later refactor - at the same time as sorting out `NodeType::pure` etc. as that should be `new_pure` or `OpType::pure(self)`, like `FunctionType::pure(self) -> Signature`)

Reply:
Good ideas, both imo! The first one, making "open_extensions" a more well supported "default" option was hampered previously by the lack of extension inference so now we can reconsider